src/asciidoctor-core-api.js

```diff
@@ -324,6 +324,114 @@ AbstractBlock.$$proto.getLineNumber = function () {
   return lineno === Opal.nil ? undefined : lineno;
 };
 
+/**
+ * Check whether this block has any child Section objects.
+ * Only applies to Document and Section instances.
+ * @memberof AbstractBlock
+ * @returns {boolean} - true if this block has child Section objects, otherwise false
+ */
+AbstractBlock.$$proto.hasSections = function () {
+  return this['$sections?']();
+};
+
+/**
+ * Get the Array of child Section objects.
+ * Only applies to Document and Section instances.
+ * @memberof AbstractBlock
+ * @returns {Array} - returns an {Array} of {@link Section} objects
+ */
+AbstractBlock.$$proto.getSections = function () {
+  return this.$sections();
+};
+
+// Section API
+
+/**
+ * @namespace
+ * @extends AbstractBlock
+ */
+var Section = Opal.Asciidoctor.Section;
+
+/**
+ * Get the 0-based index order of this section within the parent block.
+ * @memberof Section
+ * @returns {number}
+ */
+Section.$$proto.getIndex = function () {
+  return this.index;
+};
+
+/**
+ * Set the 0-based index order of this section within the parent block.
+ * @memberof Section
+ */
+Section.$$proto.setIndex = function (value) {
+  this.index = value;
+};
+
+/**
+ * Get the section name of this section.
+ * @memberof Section
+ * @returns {string}
+ */
+Section.$$proto.getSectionName = function () {
+  return this.sectname;
+};
+
+/**
+ * Set the section name of this section.
+ * @memberof Section
+ */
+Section.$$proto.setSectionName = function (value) {
+  this.sectname = value;
+};
+
+/**
+ * Get the flag to indicate whether this is a special section or a child of one.
+ * @memberof Section
+ * @returns {boolean}
+ */
+Section.$$proto.isSpecial = function () {
+  return this.special;
+};
+
+/**
+ * Set the flag to indicate whether this is a special section or a child of one.
+ * @memberof Section
+ */
+Section.$$proto.setSpecial = function (value) {
+  this.special = value;
+};
+
+/**
+ * Get the state of the numbered attribute at this section (need to preserve for creating TOC).
+ * @memberof Section
+ * @returns {boolean}
+ */
+Section.$$proto.isNumbered = function () {
+  return this.numbered;
```

Comment:
@mojavelinux `numbered` is an attribute accessor but the documentation is just "Get the state...". So I didn't add a setter. Is it OK ?

Reply:
The API doc was out of date. It should be read/write. I just updated it.